Functions/xASL_quant_BSupCalculation.m

```diff
@@ -88,18 +90,16 @@
 if PresaturationTime
     % Recovery from 0 to 1 at T1 for the period between PresaturationTime and the first BSup pulse
     SignalPercentageInitial(1) = 1 - (1 - 0)*exp(-(BackgroundSuppressionPulseTime(1)-PresaturationTime)/T1Time);
-    PresaturationTimeDummy = PresaturationTime;
 else
     % Without saturation, full signal is preset
     SignalPercentageInitial(1) = 1;
     % For the purpose of visualization, in the absence of presaturation we set the presaturation time to 1 
-    PresaturationTimeDummy = 1;
 end
  
 % First phase calculation for a graph visual output
 if ~isempty(PathGraph)
-    % Preallocate
-	SignalPercentageVector = zeros(round(ReadoutTime + max(SliceTime)), 1);
+    % Preallocate (sum is required if multiPLD and therefore multiple ReadoutTime is present)
+	SignalPercentageVector = zeros(round(sum(ReadoutTime + max(SliceTime))), 1);
```

Comment:
Are you sure it should be a sum? Is the sum on the correct location? Sum seems weird here... @jan-petr